gensim/utils.py

```diff
@@ -1263,3 +1269,45 @@ def _iter_windows(document, window_size, copy=False, ignore_below_size=True):
     else:
         for doc_window in doc_windows:
             yield doc_window.copy() if copy else doc_window
+
+
+def walk_with_depth(top, topdown=True, onerror=None, followlinks=False, depth=0):
```

Comment:
Is this really needed? The depth can be deduced easily from normal `walk()`, by comparing the root directories.